infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -171,7 +173,9 @@ public class CorfuReplicationDiscoveryService implements CorfuReplicationDiscove
      * When a non-leader node is called to do the enforcedSnapshotSync, it will write the event to
      * the shared event-table and the leader node will be notified to do the work.
      */
-    private LogReplicationEventListener logReplicationEventListener;
+    private LogReplicationEventListener logReplicationEventListener = null;
+
+    private final Map<Class, AbstractServer> serverMap = new HashMap<>();
```

Comment:
Can you rename this to something more specific?  Also, please add a comment describing what it is for

Reply:
+1

Reply:
I'm concerned why are we bringing these constructs up to the discovery service, if it is not directly receiving messages.